.github/workflows/checks.yaml

```diff
@@ -366,11 +237,9 @@ jobs:
       - image
       - integration
       - license
-      - pr
       - platform-xtest
-      - associated-issue
-    runs-on: ubuntu-latest
-    if: always()
+    runs-on: ubuntu-22.04
+    if: ${{ !cancelled() }}
```

Comment:
Yeah, I guess it is confusing to have this always be showing up as a red X everywhere, but since merges and PRs block on this we need to have it always run. If cancelled early this will count as skipped and therefore green
```suggestion
```

Reply:
I added this condition based on the docs here and @el-virt feedback. 

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/expressions#always

Reply:
If we add always. If any of the jobs it depends on are skipped this will show up as passed when having `always()`

Reply:
Can we just create two jobs? One for sensitive stuff and one for non-sensitive. 

That way our workflow isn't so complicated. It feels wrong that we're making it hard to understand because we don't want to deal with terraform and gitops.

Reply:
It looks like cancelled jobs [have the status](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/contexts#needs-context) 'cancelled'. So let's go with this and in the end merge_group will catch any problems that fall through the PR review check

also, re: naming jobs and deciding which ones are 'required' - I guess this isn't an issue for you two since you are probably both admins, but there is no way that i've found for a non-admin to know or set required jobs. So it isn't an issue with gitops but visibility. Only admins may edit which jobs are required, and non-admins get surprised with a list of 'required' jobs after generating a PR.

<img width="874" alt="image" src="https://github.com/user-attachments/assets/cdde98bb-8ec0-457c-83fc-6f26081e0af1">